lib/_stream_writable.js

```diff
@@ -134,11 +134,28 @@ Object.defineProperty(WritableState.prototype, 'buffer', {
      'instead.')
 });
 
+// Test _writableState for inheritance to account for Duplex streams,
+// whose prototype chain only points to Readable.
+var realHasInstance;
+if (Symbol.hasInstance) {
+  realHasInstance = Function.prototype[Symbol.hasInstance];
+  Object.defineProperty(Writable, Symbol.hasInstance, {
+    value: function(object) {
+      return object._writableState instanceof WritableState;
+    }
+  });
+} else {
+  realHasInstance = function(object) {
+    return object instanceof this;
+  };
+}
+
 function Writable(options) {
-  // Writable ctor is applied to Duplexes, though they're not
-  // instanceof Writable, they're instanceof Readable.
-  if (!(this instanceof Writable) && !(this instanceof Stream.Duplex))
+  // Writable ctor is applied to Duplexes, too.
+  if (!(realHasInstance.call(Writable, this)) &&
+      !(this instanceof Stream.Duplex)) {
```

Comment:
Leaving as it is will lead to infinite recursion, because `this instanceof Writable` returns `false` *before* the constructor is run, so the constructor just calls itself over and over again. And adding an extra `realHasInstance` check inside of the symbol-based override would break our fancy `LazyTransform` logic.

Feel free to play around with this, but I am afraid this would be necessary.